File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/examples-tables-attachment/examples_tables_attachment.cs
using Reqnroll;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CucumberMessages.CompatibilityTests.CCK.attachments
{
    [Binding]
    internal class Attachments
    {
        private readonly IReqnrollOutputHelper reqnrollOutputHelper;

        internal Attachments(IReqnrollOutputHelper reqnrollOutputHelper)
        {
            this.reqnrollOutputHelper = reqnrollOutputHelper;
        }

        [When(@"a JPEG image is attached")]
        public void WhenAJPEGImageIsAttached()
        {
            var ext = "jpeg" ;
            var path = FileSystemPath.GetFilePathForAttachments();
            var attachment = Path.Combine(path, "attachments", $"cucumber.{ext}");

            reqnrollOutputHelper.AddAttachment(attachment);
        }

        [When(@"a PNG image is attached")]
        public void WhenAPNGImageIsAttached()
        {
            var ext = "png";
            var path = FileSystemPath.GetFilePathForAttachments();
            var attachment = Path.Combine(path, "attachments", $"cucumber.{ext}");

            reqnrollOutputHelper.AddAttachment(attachment);
        }

    }
}

File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/examples-tables-attachment/examples_tables_attachment.ts
import { When } from '@cucumber/fake-cucumber'
import fs from 'node:fs'

When('a JPEG image is attached', async function () {
  await this.attach(fs.createReadStream(import.meta.dirname + '/cucumber.jpeg'), 'image/jpeg')
})

When('a PNG image is attached', async function () {
  await this.attach(fs.createReadStream(import.meta.dirname + '/cucumber.png'), 'image/png')
})
